dataRetriever/getters/miniBlockDataGetter.go

```diff
@@ -0,0 +1,117 @@
+package getters
+
+import (
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+var log = logger.GetOrCreate("dataretriever/getters")
+
+// ArgMiniBlockDataGetter represents a miniblock data getter
+type ArgMiniBlockDataGetter struct {
+	MiniBlockPool    storage.Cacher
+	MiniBlockStorage storage.Storer
+	Marshalizer      marshal.Marshalizer
+}
+
+// miniBlockDataGetter is able to get data from cache or from storage
+type miniBlockDataGetter struct {
+	miniBlockPool    storage.Cacher
+	miniBlockStorage storage.Storer
+	marshalizer      marshal.Marshalizer
+}
+
+// NewMiniBlockDataGetter creates a new miniBlockDataGetter instance
+func NewMiniBlockDataGetter(arg ArgMiniBlockDataGetter) (*miniBlockDataGetter, error) {
+	if check.IfNil(arg.Marshalizer) {
+		return nil, dataRetriever.ErrNilMarshalizer
+	}
+	if check.IfNil(arg.MiniBlockPool) {
+		return nil, dataRetriever.ErrNilMiniblocksPool
+	}
+	if check.IfNil(arg.MiniBlockStorage) {
+		return nil, dataRetriever.ErrNilMiniblocksStorage
+	}
+
+	return &miniBlockDataGetter{
+		miniBlockPool:    arg.MiniBlockPool,
+		miniBlockStorage: arg.MiniBlockStorage,
+		marshalizer:      arg.Marshalizer,
+	}, nil
+}
+
+// GetMiniBlocks method returns a list of deserialized mini blocks from a given hash list either from data pool or from storage
+func (mbdg *miniBlockDataGetter) GetMiniBlocks(hashes [][]byte) (block.MiniBlockSlice, [][]byte) {
```

Comment:
drop the `Get` prefix for all methods.

Reply:
I think we should keep here the Get prefix in order it does a get action not returning a stored miniblocks structure. We kind have followed this pattern in the project.

Reply:
You're right.